std/regex/internal/ir.d

```diff
@@ -899,3 +899,214 @@ struct CharMatcher {
             return trie[ch];
     }
 }
+
+// Internal non-resizeble array, switches between inline storage and CoW
+// POD-only
+struct SmallFixedArray(T, uint SMALL=3)
+if (!hasElaborateDestructor!T)
+{
+    import core.stdc.stdlib : malloc, free;
```

Comment:
How about `core.memory : pureMalloc, pureFree`? Or would there be no benefit because other stuff is impure?

Reply:
That’s a good idea. Indeed it won’t make other things pure but worth a shot.